src/buffer/out/textBuffer.cpp

```diff
@@ -2869,3 +2872,66 @@ PointTree TextBuffer::GetPatterns(const til::CoordType firstRow, const til::Coor
     PointTree result(std::move(intervals));
     return result;
 }
+
+std::vector<ScrollMark>& TextBuffer::GetMarks()
+{
+    return _marks;
+}

```

Comment:
I don't think this function is a good idea.

For instance, it leaks the way marks are stored. A `vector` is not a good way to store them and a ring buffer would be a better choice, because it makes the 2 most common operations much cheaper (adding at the end, removing at the beginning during buffer circling). Changing the underlying storage might require changing all callers around.
More importantly however I believe it's more bug-prone, because it allows the caller to make modifications that `TextBuffer` doesn't know about. Like, let's say we store marks sorted by start coordinate in the future to make range-based access and removal cheaper (`n` -> `log n`). If I change the `start` coordinate outside of `TextBuffer` then, this might break the sort order and cause the range-based functions to fail in weird ways. This might be obvious now, but I'm sure `COOKED_READ_DATA` was also obvious to someone at some point and now look at it (and us maintaining it). 😄 It has the exact same design flaw. It leaks its internals (and so does `GetRowByOffset` but I'm slowly working on fixing that).